File: WareHouse/todo_GDI_20231014012355/todo_list.py
'''
This file contains the TodoList class which manages the list of todos.
'''
import datetime
class Todo:
    def __init__(self, description):
        self.date_added = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        self.description = description
        self.date_completed = None
    def complete(self):
        self.date_completed = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
class TodoList:
    def __init__(self):
        self.todos = []
    def add_todo(self, description):
        todo = Todo(description)
        self.todos.append(todo)
    def complete_todo(self, todo):
        todo.complete()
    def get_active_todos(self):
        return [todo for todo in self.todos if todo.date_completed is None]
    def get_completed_todos(self):
        return [todo for todo in self.todos if todo.date_completed is not None]

File: WareHouse/todo_GDI_20231014012355/todolist.js
import React from 'react';
function TodoList({ todos, handleCompleteTodo }) {
  return (
    <table>
      <thead>
        <tr>
          <th>Date Added</th>
          <th>Description</th>
          {handleCompleteTodo && <th>Action</th>}
        </tr>
      </thead>
      <tbody>
        {todos.map((todo, index) => (
          <tr key={index}>
            <td>{todo.dateAdded}</td>
            <td>{todo.description}</td>
            {handleCompleteTodo && (
              <td>
                <button onClick={() => handleCompleteTodo(index)}>Complete</button>
              </td>
            )}
          </tr>
        ))}
      </tbody>
    </table>
  );
}
export default TodoList;